components/eamxx/src/physics/nudging/eamxx_nudging_process_interface.cpp

```diff
@@ -27,329 +41,289 @@ void Nudging::set_grids(const std::shared_ptr<const GridsManager> grids_manager)
-      auto v_1d = ekat::subview(v_ext,icol);
-      auto v_bef_1d = ekat::subview(v_bef,icol);
-      auto v_aft_1d = ekat::subview(v_aft,icol);
-      auto p_mid_1d = ekat::subview(p_mid_ext,icol);
-      auto p_mid_bef_1d = ekat::subview(p_mid_bef,icol);
-      auto p_mid_aft_1d = ekat::subview(p_mid_aft,icol);
-      auto qv_1d = ekat::subview(qv_ext,icol);
-      auto qv_bef_1d = ekat::subview(qv_bef,icol);
-      auto qv_aft_1d = ekat::subview(qv_aft,icol);
-
-      const auto range = Kokkos::TeamThreadRange(team, num_vert_packs);
-      Kokkos::parallel_for(range, [&] (const Int & k) {
-        T_mid_1d(k)=w_bef*T_mid_bef_1d(k) + w_aft*T_mid_aft_1d(k);
-        p_mid_1d(k)=w_bef*p_mid_bef_1d(k) + w_aft*p_mid_aft_1d(k);
-        u_1d(k)=w_bef*u_bef_1d(k) + w_aft*u_aft_1d(k);
-        v_1d(k)=w_bef*v_bef_1d(k) + w_aft*v_aft_1d(k);
-        qv_1d(k)=w_bef*qv_bef_1d(k) + w_aft*qv_aft_1d(k);
-      });
-      team.team_barrier();
-  });
-  Kokkos::fence();
-}
-
-// =========================================================================================
-void Nudging::update_time_step (const int time_s)
-{
-  //Check to see in time state needs to be updated
-  if (time_s >= m_NudgingData_aft.time)
-    {
-      const int time_index = time_s/m_time_step_file;
-      std::swap (m_NudgingData_bef,m_NudgingData_aft);
-      m_NudgingData_bef.time = m_NudgingData_aft.time;
+  m_time_interp = util::TimeInterpolation(grid_ext, m_datafiles);
 
-      m_data_input.read_variables(time_index+1);
-      Kokkos::deep_copy(m_NudgingData_aft.T_mid,m_fields_ext_h["T_mid"]);
-      Kokkos::deep_copy(m_NudgingData_aft.p_mid,m_fields_ext_h["p_mid"]);
-      Kokkos::deep_copy(m_NudgingData_aft.u,m_fields_ext_h["u"]);
-      Kokkos::deep_copy(m_NudgingData_aft.v,m_fields_ext_h["v"]);
-      Kokkos::deep_copy(m_NudgingData_aft.qv,m_fields_ext_h["qv"]);
-      m_NudgingData_aft.time = m_time_step_file*(time_index+1);
-    }
+  constexpr int ps = 1;  // TODO: I think this could be the regular packsize, right?
+  const auto& grid_name = m_grid->name();
+  if (m_src_pres_type == TIME_DEPENDENT_3D_PROFILE) {
+    create_helper_field("p_mid_ext", scalar3d_layout_mid, grid_name, ps);
+    auto pmid_ext = get_helper_field("p_mid_ext");
+    m_time_interp.add_field(pmid_ext.alias("p_mid"),true);
+  } else if (m_src_pres_type == STATIC_1D_VERTICAL_PROFILE) {
+    // Load p_lev from source data file
+    ekat::ParameterList in_params;
+    in_params.set("Filename",m_datafiles[0]);
+    in_params.set("Skip_Grid_Checks",true);  // We need to skip grid checks because multiple ranks may want the same column of source data.
+    std::map<std::string,view_1d_host<Real>> host_views;
+    std::map<std::string,FieldLayout>  layouts;
+    create_helper_field("p_mid_ext", scalar2d_layout_mid, grid_name, ps);
+    auto pmid_ext = get_helper_field("p_mid_ext");
+    auto pmid_ext_v = pmid_ext.get_view<Real*,Host>();
+    in_params.set<std::vector<std::string>>("Field Names",{"p_lev"});
```

Comment:
Sorry to bother you on this again, but the vertical remapped file uses convention `p_levs`.